backend/ibutsu_server/widgets/importance_component.py

```diff
@@ -0,0 +1,127 @@
+from ibutsu_server.constants import BARCHART_MAX_BUILDS
+from ibutsu_server.constants import JJV_RUN_LIMIT
+from ibutsu_server.db.models import Result
+from ibutsu_server.db.models import Run
+from ibutsu_server.filters import string_to_column
+from ibutsu_server.widgets.jenkins_job_view import get_jenkins_job_view
+
+
+def get_importance_component(
+    env="prod", group_field="component", job_name="", builds=5, components=None, project=None
+):
+    # taken from get_jenkins_line_chart in jenkins_job_analysis.py
+    run_limit = int((JJV_RUN_LIMIT / BARCHART_MAX_BUILDS) * builds)
+    jobs = get_jenkins_job_view(
+        filter_=f"job_name={job_name}", page_size=builds, project=project, run_limit=run_limit
+    ).get("jobs")
+
+    # A list of job build numbers to filter our runs by
+    job_ids = []
+    for job in jobs:
+        job_ids.append(job["build_number"])
+
+    # query for RUN ids
+    # metadata has to have a string to column to work
+    # because it is a sqlalchemy property otherwise (AFAIK)
+    bnumdat = string_to_column("metadata.jenkins.build_number", Run)
+    run_data = (
+        Run.query.filter(bnumdat.in_(job_ids), Run.component.in_(components.split(",")))
```

Comment:
If I send a request to the backend locally  `curl http://localhost:8080/api/widget/importance-component -H "Accept: application/json" -H "Authorization: Bearer $IBUTSU_TOK"`

it fails exaclty on this place:
```
Traceback (most recent call last):
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask/app.py", line 2552, in __call__
    return self.wsgi_app(environ, start_response)
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask/app.py", line 2532, in wsgi_app
    response = self.handle_exception(e)
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask_cors/extension.py", line 176, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask_cors/extension.py", line 176, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "/mnt/.backend_env/lib/python3.9/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/mnt/.backend_env/lib/python3.9/site-packages/connexion/decorators/decorator.py", line 68, in wrapper
    response = function(request)
  File "/mnt/.backend_env/lib/python3.9/site-packages/connexion/security/security_handler_factory.py", line 388, in wrapper
    return function(request)
  File "/mnt/.backend_env/lib/python3.9/site-packages/connexion/decorators/uri_parsing.py", line 149, in wrapper
    response = function(request)
  File "/mnt/.backend_env/lib/python3.9/site-packages/connexion/decorators/validation.py", line 399, in wrapper
    return function(request)
  File "/mnt/.backend_env/lib/python3.9/site-packages/connexion/decorators/parameter.py", line 120, in wrapper
    return function(**kwargs)
  File "/mnt/ibutsu_server/controllers/widget_controller.py", line 103, in get_widget
    return WIDGET_METHODS[id_](**params)
  File "/mnt/ibutsu_server/widgets/importance_component.py", line 28, in get_importance_component
    Run.query.filter(bnumdat.in_(job_ids), Run.component.in_(components.split(",")))
AttributeError: 'NoneType' object has no attribute 'split'

```

Reply:
Oh, I wonder if that is because components aren't required

Reply:
@LightOfHeaven1994 I changed the default to something that won't fail on a split